R/derive_param_bor.R

```diff
@@ -0,0 +1,518 @@
+#'   "2",      "2020-01-01", "SD",
+#'   "2",      "2020-02-01", "PR",
+#'   "2",      "2020-03-01", "SD",
+#'   "2",      "2020-03-13", "CR",
+#'   "3",      "2019-11-12", "CR",
+#'   "3",      "2019-12-02", "CR",
+#'   "3",      "2020-01-01", "SD",
+#'   "4",      "2020-01-01", "PR",
+#'   "4",      "2020-03-01", "SD",
+#'   "4",      "2020-04-01", "SD",
+#'   "4",      "2020-05-01", "PR",
+#'   "4",      "2020-05-15", "NON-CR/NON-PD",
+#'   "5",      "2020-01-01", "PR",
+#'   "5",      "2020-01-10", "SD",
+#'   "5",      "2020-01-20", "PR",
+#'   "5",      "2020-05-15", "NON-CR/NON-PD",
+#'   "6",      "2020-02-06", "PR",
+#'   "6",      "2020-02-16", "CR",
+#'   "6",      "2020-03-30", "PR",
+#'   "6",      "2020-04-12", "PD",
+#'   "6",      "2020-05-01", "CR",
+#'   "6",      "2020-06-01", "CR",
+#'   "7",      "2020-02-06", "PR",
+#'   "7",      "2020-02-16", "CR",
+#'   "7",      "2020-04-01", "NE"
+#' ) %>%
+#'   mutate(PARAMCD = "OVR")
+#'
+#' pd_obs <-
+#'   bind_rows(tribble(
+#'     ~USUBJID, ~ADTC,        ~AVALC,
+#'     "2",      "2020-03-01", "Y",
+#'     "4",      "2020-02-01", "Y"
+#'   ) %>%
+#'     mutate(PARAMCD = "PD"))
+#'
+#' adrs <- bind_rows(ovr_obs, pd_obs) %>%
+#'   mutate(
+#'     ADT = ymd(ADTC),
+#'     STUDYID = "XX1234"
+#'   ) %>%
+#'   select(-ADTC) %>%
+#'   derive_vars_merged(
+#'     dataset_add = adsl,
+#'     by_vars     = vars(STUDYID, USUBJID),
+#'     new_vars    = vars(TRTSDT)
+#'   )
+#'
+#' pd_date <- date_source(
+#'   dataset_name = "adrs",
+#'   date         = ADT,
+#'   filter       = PARAMCD == "PD"
+#' )
+#'
+#' aval_fun_pass <- function(arg) {
+#'   case_when(
+#'     arg == "CR" ~ 11,
+#'     arg == "PR" ~ 22,
+#'     arg == "SD" ~ 33,
+#'     arg == "NON-CR/NON-PD" ~ 44,
+#'     arg == "PD" ~ 55,
+#'     arg == "NE" ~ 66,
+#'     arg == "MISSING" ~ 77,
+#'     TRUE ~ NA_real_
+#'   )
+#' }
+#'
+#' # Derive best overall response parameter
+#' derive_param_bor(
+#'   adrs,
+#'   dataset_adsl = adsl,
+#'   filter_source = PARAMCD == "OVR",
+#'   source_pd = pd_date,
+#'   source_datasets = list(adrs = adrs),
+#'   aval_fun = aval_fun_pass,
+#'   reference_date = TRTSDT,
+#'   ref_start_window = 28,
+#'   set_values_to = vars(
+#'     PARAMCD = "BOR",
+#'     PARAM = "Best Overall Response"
+#'   )
+#' ) %>%
+#'   filter(PARAMCD == "BOR")
+#' @export
+#
+#' @author Stephen Gormley
+#
+#' @keywords ADRS
+#
+#' @return The dataframe passed in the `dataset` argument with additional columns and/or
+#'         rows as set in the `set_values_to` argument.
+
+derive_param_bor <- function(dataset,
+                             dataset_adsl,
+                             filter_source,
+                             source_pd = NULL,
+                             source_datasets = NULL,
+                             reference_date,
+                             ref_start_window = 0,
+                             missing_as_ne = FALSE,
+                             aval_fun = aval_resp(),
```

Comment:
Please remove the brackets. The default value should be the function not the call of the function.

Reply:
Updated, in my test I pass in a function so did not notice, updated test to include default call.